roles/aap_ocp_install/tasks/main.yml

```diff
@@ -32,6 +34,8 @@
     apply:
       tags:
         - controller
+  when:
+    - ( 'controller' in ansible_run_tags ) or ( 'all' in ansible_run_tags and aap_ocp_install_controller is defined )
```

Comment:
should be sufficient for the If defined, if someones running all it will do the right thing, if they are only running one tag or the other and the variable, it will do the right thing. No need to check the tags in the when. I think they take care of themselves.

Reply:
I simplified the when conditions to only look for the variable to be defined. :man_facepalming: 

I tested with various combinations of variables defined and specifying tags.